src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    static class Node<T> {
+        private Node<T> prev;
+        private T item;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T items, Node<T> next) {
+            this.prev = prev;
+            this.item = items;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            addLast(value);
+        } else {
+            addBefore(value, nodeOnIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T data : list) {
+            add(data);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkForIndex(index);
+        return nodeOnIndex(index) == null ? null : nodeOnIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkForIndex(index);
+        T oldValue = nodeOnIndex(index).item;
+        nodeOnIndex(index).item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkForIndex(index);
+        return unlink(nodeOnIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> nodeEdited = first; nodeEdited != null; nodeEdited = nodeEdited.next) {
+            if (nodeEdited.item == null && object == null
```

Comment:
```suggestion
            if (nodeEdited.item ==  object
```
тут можна спростити перевірку, так як нас цікавить лише випадок, коли обидва значення нал